### PR TITLE
Make clang checks of bad `--ccflags` fatal

### PR DESCRIPTION
**Pull request by bradcray**
[designed by @riftEmber, typed up by me]

Prior to this PR, if the LLVM back-end's processing of `--ccflags` turned up any bad/unexpected flags, an error would be printed, but the Chapel compiler would continue.  As an example of that, the existing `unknownLlvmFlag.chpl` test would issue the LLVM error, then complete compilation, permitting the program to be run.  Here, we detect the presence of the the error when asking `clang` to process `--ccflags`, and halt the Chapel compiler as a result, which seems appropriate for an erroneous condition.

This change was motivated by the fact that the `--compiler-driver` mode invokes this check twice, so was generating the error message twice.  By having the first case halt after generating an error, both modes now behave the same and the error message is only printed once.

Beyond the change itself, the only other change required was to update the only test we seem to have so far that generates an error in the LLVM back-end to not include the output of the executable and add the new error message.  Surprisingly, no other tests needed updating in a comm=none paratest run.

### CODE REVIEW

**Comment by bradcray** [comment]
@mppf: This is short and sweet and in the middle of some code you developed (or at least, last touched?).  Would you have time to give it a review?

Size: 1/7
Complexity: 4/7
Pride: 7/7 (though all of it goes to Anna)

File: compiler/llvm/clangUtil.cpp [hunk 1]
```diff
@@ -1701,6 +1701,10 @@ void setupClang(GenInfo* info, std::string mainFile)
   std::unique_ptr<clang::driver::Compilation> C(
       TheDriver.BuildCompilation(clangInfo->driverArgsCStrings));
 
+  if (diagClient->getNumErrors() > 0) {
+    USR_FATAL("error in back-end LLVM cmopilation");
```
Comment (mppf) [inline compiler/llvm/clangUtil.cpp:1705]: spalling: cmopilation

**Review by mppf — APPROVED** [review]
Thanks for fixing this - I'd definitely want compilation to halt if the LLVM backend encounters an error. The fix looks reasonable to me other than the spelling error.

**Comment by mppf** [comment]
Also, in terms of communicating this change -- what you have here doesn't make errors in LLVM passes fatal. It's just making errors in the arguments to the `clang` we are integrating with fatal. So it is really about errors in `--ccflags`. It's still a good change, but it'd be good to clarify this in the PR title / PR message.

It might be good to also add a test for `--mllvm` flags that don't exist to make sure those are fatal. This seems OK already:

```
$ chpl hello.chpl  --mllvm -nonexistent-flag
chpl --mllvm: Unknown command line argument '-nonexistent-flag'.  Try: 'chpl --mllvm --help'
chpl --mllvm: Did you mean '--insert-all0'?
$ ls hello
ls: cannot access 'hello': No such file or directory
```

**Review by riftEmber — APPROVED** [review]

**Comment by bradcray** [comment]
@mppf —

> It might be good to also add a test for --mllvm flags that don't exist to make sure those are fatal.

I may be misunderstanding, but doesn't `compflags/ferguson/mllvm-badarg.chpl` do this already (in that the test contains a writeln() that doesn't seem to get printed out)?

**Comment by bradcray** [comment]
> Also, in terms of communicating this change...

Thanks for correcting my understanding.  I've updated the title and OP to try and make it more accurate.  Would one of you (@mppf or @riftEmber) make sure I've got it right now?  Thanks!

**Comment by riftEmber** [comment]
>  I've updated the title and OP to try and make it more accurate.

LGTM

**Comment by mppf** [comment]
> I may be misunderstanding, but doesn't `compflags/ferguson/mllvm-badarg.chpl` do this already (in that the test contains a writeln() that doesn't seem to get printed out)?

Yes, that covers it, thanks for finding it & verifying.

**Pull request merged**